### PR TITLE
Fixing Intellij Bug to support socks proxy edge case

### PR DESCRIPTION
**Pull request by arafatkatze**
## Test plan
Tested in development locally

### CODE REVIEW

File: src/main/kotlin/com/sourcegraph/cody/agent/CodyAgent.kt [hunk 1]
```diff
@@ -172,7 +172,7 @@ private constructor(
       if (proxy.PROXY_TYPE_IS_SOCKS) {
         processBuilder.environment()["HTTP_PROXY"] = "socks://$proxyUrl"
       }
-      if (proxy.USE_HTTP_PROXY) {
+      else {
```
Comment (pkukielka) [inline src/main/kotlin/com/sourcegraph/cody/agent/CodyAgent.kt]: I do not understand this change, old code looks valid to me and proposed change seems to be incorrect.
Could you please elaborate on the proposed logic change?

File: src/main/kotlin/com/sourcegraph/cody/agent/CodyAgent.kt [hunk 2]
```diff
@@ -172,7 +172,7 @@ private constructor(
       if (proxy.PROXY_TYPE_IS_SOCKS) {
```
Comment (pkukielka) [inline src/main/kotlin/com/sourcegraph/cody/agent/CodyAgent.kt]: That code is not correct. If you switch proxy type to socks, and then disable proxy, `PROXY_TYPE_IS_SOCKS` is still set to `true` (I checked in debugger), but it should not be passed to agent.
Correct implementation would be:

```kt
 if (proxy.USE_HTTP_PROXY) {
    if (proxy.PROXY_TYPE_IS_SOCKS) {
        processBuilder.environment()["HTTP_PROXY"] = "socks://$proxyUrl"
    } else {
        processBuilder.environment()["HTTP_PROXY"] = "http://$proxyUrl"
        processBuilder.environment()["HTTPS_PROXY"] = "http://$proxyUrl"
    }
}

**Review by pkukielka — APPROVED** [review]
LGTM
